pkg/model/multiversion/delta.go

```diff
@@ -0,0 +1,248 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package multiversion
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	awssdkmodel "github.com/aws/aws-sdk-go/private/model/api"
+
+	ackmodel "github.com/aws-controllers-k8s/code-generator/pkg/model"
+)
+
+// FieldChangeType represent the type of field modification.
+type FieldChangeType string
+
+const (
+	// FieldChangeTypeUnknown is used when ChangeType cannot be computed.
+	FieldChangeTypeUnknown FieldChangeType = "unknown"
+	// FieldChangeTypeIntact is used a field name and structure didn't change.
+	FieldChangeTypeIntact FieldChangeType = "intact"
+	// FieldChangeTypeAdded is used when a new field is introduced in a CRD.
+	FieldChangeTypeAdded FieldChangeType = "added"
+	// FieldChangeTypeRemoved is used a when a field is removed from a CRD.
+	FieldChangeTypeRemoved FieldChangeType = "removed"
+	// FieldChangeTypeRenamed is used when a field is renamed.
+	FieldChangeTypeRenamed FieldChangeType = "renamed"
+	// FieldChangeTypeShapeChanged is used when a field shape has changed.
+	FieldChangeTypeShapeChanged FieldChangeType = "shape-changed"
+	// FieldChangeTypeShapeChangedToSecret is used when a field change to
+	// a k8s secret type.
+	FieldChangeTypeShapeChangedToSecret FieldChangeType = "shape-changed-to-secret"
```

Comment:
We need discrete types for secret?

Reply:
I believe that we need to handle `string` to `secret` changes differently than other type changes